lib/phoenix_html/link.ex

```diff
@@ -135,13 +135,18 @@ defmodule Phoenix.HTML.Link do
 
   defp csrf_data(opts) do
     {csrf_token?, opts} = Keyword.pop(opts, :csrf_token, true)
-    if csrf_token = csrf_token? && Plug.CSRFProtection.get_csrf_token() do
+    if csrf_token = csrf_token? && get_csrf_token() do
       {[csrf: csrf_token], opts}
     else
       {[], opts}
     end
   end
 
+  defp get_csrf_token do
+    generator = Application.get_env(:phoenix_html, :csrf_token_generator, Plug.CSRFProtection)
```

Comment:
I think it would be better if it was a MFA (module-function-args):

    {mod, fun, args} = Application.fetch_env(:phoenix_html, :csrf_token_generator)
    apply(mod, fun, args)

Also note that you don't need a default here. Instead, in the "mix.exs" file under `def application`, add this:

    env: {Plug.CSRFProtection, :get_csrf_token, []}

and now you can expect it to always be available.